barcodes_generator_abstract/models/barcode_rule.py

```diff
@@ -54,6 +61,33 @@ def onchange_generate_type(self):
             if rule.generate_type == 'no':
                 rule.generate_model = False
 
+    # Constrains Section
+    @api.multi
+    @api.constrains('generate_model', 'generate_automate')
+    def _check_generate_model_automate(self):
+        """ It should not allow two automated barcode generators per model.
+
+        It also clears the cache of automated rules if necessary.
+        """
+        should_clear = False
+        for record in self:
+            if not record.generate_automate:
+                continue
+            # This query is duplicated, but necessary because the other
+            # method is cached & we need a completely current result.
+            domain = [
+                ('generate_model', '=', record.generate_model),
+                ('generate_automate', '=', True),
+            ]
+            if len(self.search(domain)) > 1:
+                raise exceptions.ValidationError(_(
+                    'Only one rule per model can be used for automatic '
+                    'barcode generation.'
+                ))
+            should_clear = True
+        if should_clear:
+            self.clear_caches()
```

Comment:
Hmmmm yeah that would work too, although then we'd be splitting the logic into two spots & it would need to be a conjoined if - because this is also clearing cache is `generate_model` is changed too.

I could remove the somewhat ghetto `should_clear` as `if len(self)` instead though, which would remove two lines and be the same level of efficiency.

I do somewhat take reservation to a mutating operation in a constraint though, so it might still be best to bite the bullet and move into the two spots. Any additional thoughts?